vignettes/BiobaseDevelopment.Rmd

```diff
@@ -0,0 +1,510 @@
+
+```{r}
+setMethod("initialize", "MySet",
+          function(.Object, ...) {
+            .Object <- callNextMethod(.Object, ...) 
+            }) 
+```
+
+and manipulated before being returned to the user. Finally, it is the responsibility of the developer to ensure that a valid object is created; `callNextMethod` is a useful way to exploit correctly designed `initialize` methods for classes that the object extends, but the developer is free to use other techniques to create valid versions of their class.
+
+A validity method might complete our new class. A validity method is essential to ensure that the unique features of *SwirlSet* -- our reason for designing the new class -- are indeed present. We define our validity method to ensure that the `assayData` slot contains our four types of expression elements:
+
+```{r}
+setValidity("SwirlSet", function(object) {
+  assayDataValidMembers(assayData(object), c("R", "G", "Rb", "Gb"))
+})
+```
+
+Notice that we do not have to explicitly request that the validity of other parts of the *SwirlSet* object are valid; this is done for us automatically. Objects are checked for validity when they are created, but not when modified. This is partly for efficiency reasons, and partly because object updates might transiently make them invalid. So a good programing practice is to ensure validity after modification, e.g.,
+
+```{r eval=FALSE}
+myFancyFunction <- function(obj) {
+  assayData(obj) <- fancyAssaydData # obj invalid...
+  phenoData(obj) <- justAsFancyPhenoData # but now valid 
+  validObject(obj) 
+  (obj)
+}
+```
+
+Assigning `fancyAssaydData` might invalidate the object, but `justAsFancyPhenoData` restores validity.
+
+# *Versioned*
+
+One problem encountered in the Bioconductor project is that data objects stored to disk become invalid as the underlying class definition changes. For instance, earlier releases of *Biobase* contain a sample *eSet* object. But under the changes discussed here, `eSet` is virtual and the stored object is no longer valid. The challenge is to easily identify invalid objects, and to provide a mechanism for updating old objects to their new representation.
+
+*Biobase* introduces the *Versioned* and *VersionedBiobase* classes to facilitate this. These classes are incorporated into key *Biobase*class definitions.*Biobase* also defines `updateObject` methods (the `updateObject` generic function is defined in the *BiocGenerics* package) for conveniently updating old objects to their new representation.
```

Comment:
Let's make `BiocGenerics` a link to the package with `Biocpkg("BiocGenerics")`

Reply:
Do you want me to adda hyperlink and link it to package installation page?

Reply:
Since we're using BiocStyle, the function `Biocpkg()` will make the link for us. (Sorry that wasn't clear!)

Reply:
Ok got it , thanks
Also one for thing, I have made multiple commits here. Is it okay as is or I should squash them to keep the history clean?

Reply:
I will squash commits when I merge.  For the link, try using `r Biocpkg("BiocGenerics")`. Here's the BiocStyle documentation: https://bioconductor.org/packages/release/bioc/vignettes/BiocStyle/inst/doc/AuthoringRmdVignettes.html#4_Style_macros.

Reply:
Done and pushed changes.
Thank you